filecoin-proofs/src/api/post.rs

```diff
@@ -460,15 +463,16 @@ pub fn generate_fallback_sector_challenges<Tree: 'static + MerkleTreeTrait>(
     let mut sector_challenges: BTreeMap<SectorId, Vec<u64>> = BTreeMap::new();
 
     let num_sectors_per_chunk = post_config.sector_count;
-    let partitions = if post_config.typ == PoStType::Window {
-        match get_partitions_for_window_post(pub_sectors.len(), &post_config) {
+    let partitions = match post_config.typ {
+        PoStType::Window => match get_partitions_for_window_post(pub_sectors.len(), &post_config) {
             Some(x) => x,
             None => 1,
-        }
-    } else {
-        assert_eq!(post_config.typ, PoStType::Winning);
+        },
+        PoStType::Winning => {
+            assert_eq!(post_config.typ, PoStType::Winning);
```

Comment:
This assertion is quite redundant here. The previous line literally guarantees this will never fail, so I think it can be removed. I would instead add a comment justifying the decision. Since I don't think the number of partitions is _structurally_ guaranteed to be one, I would probably add a comment justifying this decision. That comment should make clear what change to the code base (even if it can't be conveniently checked here) would necessitate a change here.

Reply:
Agreed, that looks like a hold-over from the previous if/else branch change.  Rather than update and push, I'm opting to merge this and update in a future PR.